org.eclipse.scanning.api/src/org/eclipse/scanning/api/ui/IResultHandler.java

```diff
@@ -22,4 +24,13 @@
 	 * @return true if result open handled ok, false otherwise. Normally if the user chooses not to proceed true is still returned.
 	 */
 	public boolean open(T bean) throws Exception;
+	
+	default public String getErrorMessage(T bean) {
+		return MessageFormat.format("Cannot open {0} normally.\n\nPlease contact your support representative.", getLocation(bean));
+	}
+	
+	default public String getLocation(T bean) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)